api/cluster-config/v1/clusterconfig_types.go

```diff
@@ -0,0 +1,57 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ClusterConfigSpec defines the desired state of ClusterConfig
+type ClusterConfigSpec struct {
```

Comment:
Question: is it possible to set an maxLimit in this struct or in the crd to say that the max possible value is 100%?

Reply:
https://book.kubebuilder.io/reference/markers/crd-validation.html 
```kubebuilder:validation:Maximum
kubebuilder:validation:Minimum
```